### ISSUE TITLE
Upgradation Results Case

### ISSUE THREAD

**Issue by gauravjain2** [ankushgarg1998/ipu-ranklist#81]
This doesn't properly take case of students who got upgraded in overall result.

Let's Say Student 'A' did B.Tech in ECE in 1st and 2nd Sem and then got upgraded to CSE in 3rd Sem. Then, in this case, the website is showing overall percentage counting from sem 3 onwards and not counting the sem 1 and 2 result.

Issue you will face is that enrollment number of student changes after upgrading but the result for earlier semester is carried forward.

**Comment by ankushgarg1998** [comment]
Right. I don't know how you're listening in on my conversation with @Eshaan7 😂 but you're raising the same issues.

I've had this problem since day one. Was trying to ignore it for as long as possible 😛 
The issue here is that exactly what you pointed out in the last statement.

**Comment by ankushgarg1998** [comment]
Just the fact that we do not have any publicly available mapping of connecting enrollment numbers, makes this problem theoretically impossible to solve.

**Comment by ankushgarg1998** [comment]
The only way out of this is an option that allows a user to enter both his enrollment numbers, and shows him his individual collated result.

As discussed in #43

**Comment by Eshaan7** [comment]
> The only way out of this is an option that allows a user to enter both his enrollment numbers, and shows him his individual collated result.
> 
> As discussed in #43

Yep. Often the best solution is: no overengineering :D.

**Comment by divyansh000915** [comment]
I am one of  USICT (Formerly USIT, consider changing college name too) students upgraded (intra college) (2022 batch) in 2019 (after 1st year) 

Here's the list of upgraded students of my batch, Older Roll Nos. can be found below, For New Roll Nos., you may try searching in the new branch with the student's name, phone no. or like a pool of comparisons or some other way. Another idea (for USICT UPGRADATION cases) is to check for missing student records in the 3rd sem (4th sem preferably) lists and compare with other branch records for that semseter [Just giving a clue]
http://ipu.ac.in/Pubinfo2019/upgrade040919.PDF

**Comment by gauravjain2** [comment]
About the public mapping
Each year university releases such mapping on the website.

Here is an example from session 2018-19 [Revised Enroll No](http://ipu.ac.in/pubinfo/Upgradation161018.PDF)

But that's a scanned copy and I don't think it can be directly parsed
And would take up a lot of manual work, though.

**Comment by ankushgarg1998** [comment]
Wow 😂 
I guess we've established I know very little about my own university. I had no clue this info was publicly available.

Having said that, that's about 200 manual entries. Not scalable at all.

**Comment by ankushgarg1998** [comment]
Unless some Cybersecurity guy helps me hack the actual PDFs/Excel sheets from IPU. 🌚 
Nudge Nudge @Eshaan7

**Comment by ankushgarg1998** [comment]
> I am one of USICT (Formerly USIT, consider changing college name too) students upgraded (intra college) (2022 batch) in 2019 (after 1st year)
> 
> Here's the list of upgraded students of my batch, Older Roll Nos. can be found below, For New Roll Nos., you may try searching in the new branch with the student's name, phone no. or like a pool of comparisons or some other way. Another idea (for USICT UPGRADATION cases) is to check for missing student records in the 3rd sem (4th sem preferably) lists and compare with other branch records for that semseter [Just giving a clue]
> http://ipu.ac.in/Pubinfo2019/upgrade040919.PDF

@divyansh000915 Thanks for bringing this up. I've given the solution, you're suggesting a fair amount of thought.

Let's say I set-up a system to match enrollment numbers with only 1st 2nd sem result and corresponding 3-8th sem result. Let's say I even add an admission year consistency check over the same "name" check too.
This is not enough to conclude that these two can be joined (without any conflicts).

**Comment by Eshaan7** [comment]
> Having said that, that's about 200 manual entries. Not scalable at all.

So about 200~ and that too each year!

> Unless some Cybersecurity guy helps me hack the actual PDFs/Excel sheets from IPU. :new_moon_with_face:

I stick to the original solution [ref](https://github.com/ankushgarg1998/ipu-ranklist/issues/81#issuecomment-767100264). xD
One thing to make it easier for such students would be an URL pattern like this perhaps:
```
https://www.ipuranklist.com/migration-student/<roll_no_1>/<roll_no_2>
```

**Comment by ankushgarg1998** [comment]
> So about 200~ and that too each year!
🤯 

> I stick to the original solution [ref](https://github.com/ankushgarg1998/ipu-ranklist/issues/81#issuecomment-767100264). xD
😆 

> One thing to make it easier for such students would be an URL pattern like this perhaps:
🔥

**Comment by ankushgarg1998** [comment]
#43 is resolved now.

For all the migration and upgradation students:
https://www.ipuranklist.com/upgradation-student

Go Nuts!

**Comment by ankushgarg1998** [comment]
### RECENT FINDING

There's something called "Student ID". It remains the same across  multiple enrollment numbers of a student for the following cases:
- Changing branch (Upgradation)
- Changing college (Migration)

This problem of linking two enrollment numbers might actually be solvable!
